gpflux/models/gi_deep_gp.py

```diff
@@ -0,0 +1,418 @@
+#
+# Copyright (c) 2021 The GPflux Contributors.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+#
+"""
+This module provides an implementation of global inducing points for deep GPs. See Ober &
+Aitchison (2021) https://arxiv.org/abs/2005.08140
+
+Note: this is a sample-based implementation, so we return samples from the DGP instead of predictive
+means and variances.
+"""
+
+import itertools
+from typing import List, Optional, Tuple, Type
+
+import tensorflow as tf
+import tensorflow_probability as tfp
+
+from gpflow import Parameter, default_float
+from gpflow.base import Module, TensorType
+
+import gpflux
+from gpflux.layers import LayerWithObservations, SampleBasedGaussianLikelihoodLayer
+
+
+class GIDeepGP(Module):
```

Comment:
There appears to be a lot of code duplication from `DeepGP`. Can you not inherit from `DeepGP` and overwrite the appropriate methods?